x-pack/legacy/plugins/maps/public/layers/sources/es_geo_grid_source/create_source_editor.js

```diff
@@ -116,7 +116,9 @@ export class CreateSourceEditor extends Component {
     });
 
     //make default selection
-    const geoFields = indexPattern.fields.filter(filterGeoField);
+    const geoFields = indexPattern.fields
+      .filter(field => !isNestedField(field))
```

Comment:
Looks like filterGeoField callback is used in [SingleFieldSelect](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/components/single_field_select.js#L27) to filter the fields list. There needs to be some clean-up since the filtering in SingleFieldSelect is not really needed since all consumers pre-filter the list anyways. That clean-up work is outside the scope of your PR but you will need to ensure filterGeoFieldfilters out nested fields until the cleanup can be completed